packages/react-kit/src/components/Header/Header.tsx

```diff
@@ -0,0 +1,145 @@
+import { IconButton } from '@components/Button';
+import clsx from 'clsx';
+import FocusTrap from 'focus-trap-react';
+import React, { useEffect, useState } from 'react';
+import { CloseSVG, HamburgerMenuSVG } from './HeaderIcons';
+import { NavHeaderProps } from './props';
+
+/**
+ * @see {@link https://live-storybook-react-kit.appa.pantheon.site/?path=/docs/header-header--docs}
+ */
+export const Header = ({
+	Logo,
+	mainNavItems,
+	secondaryNavItems,
+	overlayStyles,
+	navbarStyles,
+	focusTrapOptions,
+}: NavHeaderProps) => {
+	const [isOpen, setIsOpen] = useState(false);
+	const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+	const handleOpen = () => setIsOpen((prev) => !prev);
+
+	const MAX_MOBILE_NAV_WIDTH = 1023;
+
+	useEffect(() => {
+		const handleKeyUp = (e: KeyboardEvent) => {
+			if (e.key === 'Escape' || e.key === 'Esc') {
+				setIsOpen(false);
+			}
+		};
+		const handleResize = () => {
+			setWindowWidth(window.innerWidth);
+			windowWidth > MAX_MOBILE_NAV_WIDTH && setIsOpen(false);
+		};
+		setWindowWidth(window.innerWidth);
+
+		window.addEventListener('resize', handleResize);
+		window.addEventListener('keyup', handleKeyUp);
+
+		return () => {
+			window.removeEventListener('resize', handleResize);
+			window.removeEventListener('keyup', handleKeyUp);
+		};
+	}, [windowWidth]);
+
+	/**
+	 * Need to use a ref here so the FocusTrap can work properly.
+	 */
+	const NavList = React.forwardRef<
+		React.ElementRef<'nav'>,
+		Pick<NavHeaderProps, 'mainNavItems' | 'secondaryNavItems'>
+	>(({ mainNavItems, secondaryNavItems }, ref) => {
+		const NAVLIST_STYLES = {
+			nav: clsx('rk-flex rk-items-center'),
+			ul: clsx(
+				'lg:rk-menu-horizontal',
+				isOpen && windowWidth <= MAX_MOBILE_NAV_WIDTH
+					? 'rk-menu-vertical rk-w-full'
+					: 'rk-hidden rk-flex-wrap lg:rk-flex',
+			),
+		};
+
+		return (
+			<header
+				ref={ref}
+				className={clsx(
+					navbarStyles,
+					'rk-z-10 rk-flex rk-w-full rk-items-center rk-py-4',
+				)}
+			>
+				<Logo />
```

Comment:
This is a smaller (and thus lower priority) case of putting too much markup responsibility on the consumer as discussed below in the comment about the nav items. If we come up with a way to solve that using slots, I wonder if that same approach could apply here. If not, that is probably OK - markup here is less complicated, and less likely to change.

Reply:
Are you referring to the `Logo` component here? GitHub is making it hard to tell.

Reply:
Yes - was referring to the logo here.